src/vue/forms/IssuanceForm.vue

```diff
@@ -0,0 +1,234 @@
+        :disabled="formMixin.isDisabled"
+        @click.prevent="cancelForm"
+      >
+        {{ 'issuance.cancel-btn' | globalize }}
+      </button>
+    </div>
+  </form>
+</template>
+
+<script>
+import IssuanceFormMixin from '@/vue/mixins/issuance-form.mixin'
+
+import { Bus } from '@/js/helpers/event-bus'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+
+import { Sdk } from '@/sdk'
+import { base } from '@tokend/js-sdk'
+
+import { required, amountRange } from '@validators'
+
+export default {
+  name: 'issuance-form',
+  mixins: [IssuanceFormMixin],
+  data: _ => ({
+    form: {
+      asset: '',
+      amount: 0,
+      receiver: '',
+      reference: ''
+    }
+  }),
+  validations () {
+    return {
+      form: {
+        asset: { required },
+        amount: {
+          required,
+          amountRange: amountRange(
+            0.000001,
+            this.availableAmount ? this.availableAmount.value : 0
+          )
+        },
+        receiver: { required },
+        reference: { required }
+      }
+    }
+  },
+  computed: {
+    assetListValues () {
+      return this.ownedAssets
+        .map(asset => ({
+          value: asset.code,
+          label: `${asset.details.name} (${asset.code})`
+        }))
+    },
+    availableAmount () {
+      if (this.ownedAssets.length && this.form.asset) {
+        const asset = this.ownedAssets
+          .find(asset => asset.code === this.form.asset)
+        return {
+          value: asset.availableForIssuance,
+          currency: asset.code
+        }
+      }
+      return { value: 0 }
+    }
+  },
+  async created () {
+    await this.loadAssets()
+    if (this.ownedAssets.length > 0) {
+      this.form.asset = this.ownedAssets[0].code
+    }
+  },
+  methods: {
+    async submit () {
+      if (!this.isFormValid()) return
+      this.disableForm()
+      try {
+        const receiverBalance =
+          await this.getReceiverBalance(this.form.receiver, this.form.asset)
+
+        if (receiverBalance) {
+          const operation =
+            base.CreateIssuanceRequestBuilder.createIssuanceRequest({
+              asset: this.form.asset,
+              amount: this.form.amount.toString(),
+              receiver: receiverBalance.balanceId,
+              reference: this.form.reference,
+              externalDetails: {}
+            })
+          await Sdk.horizon.transactions.submitOperations(operation)
+          Bus.success('issuance.tokens-issued-msg')
+          this.cancelForm()
+        }
+      } catch (e) {
+        console.error(e)
+        ErrorHandler.process(e)
+      }
+      this.enableForm()
+    },
+    async getAccountInfoByEmail (email) {
+      const response = await Sdk.api.users.getPage({ email: email })
+      const account = response.data.find(info => info.email === email)
+      if (account) return account
+      Bus.error('issuance.wrong-email-err')
+      return null
+    },
+    async getReceiverBalance (receiver, asset) {
```

Comment:
same here: do not return different types

Reply:
Resolved, updated